shell/packages/sandstorm-db/db.js

```diff
@@ -1797,6 +1789,50 @@ _.extend(SandstormDb.prototype, {
     return setting && setting.value && this.isFeatureKeyValid();
   },
 
+  updateUserQuotaFromLdap: function (user) {
+    const fallback = user.cachedStorageQuota || 0;
+    if (Meteor.isClient) return fallback;
+
+    const setting = this.collections.settings.findOne({ _id: "quotaLdapAttribute" });
+    if (!setting) return fallback;
+
+    const ldap = new LDAP();
+
+    // TODO(someday): don't just assume the first login identity is the primary identity?
+    const email = this.getPrimaryEmail(user._id, user.loginIdentities[0].id);
+    if (!email) return fallback;
+
+    const ldapUser = ldap.ldapCheck(this, { searchUsername: email, });
+    if (!ldapUser || !ldapUser.searchResults) return fallback;
+
+    const newStorageQuota = +ldapUser.searchResults[setting.value];
+    Meteor.users.update({ _id: user._id }, { $set: { cachedStorageQuota: newStorageQuota } });
+    // TODO(someday): cache timestamp as well and only check/update if greater than 60s ago
+    return newStorageQuota || fallback;
+  },
+
+  getUserQuota: function (user) {
+    if (this.isReferralEnabled()) {
+      // TODO(someday): isReferralEnabled isn't quite the right check
```

Comment:
I think what you want to be checking is "is LDAP quota enabled?", not "is Oasis?"